public/app/features/alerting/components/AlertingQueryPreview.tsx

```diff
@@ -1,12 +1,16 @@
-import React, { FC, useMemo, useState } from 'react';
-import { useObservable } from 'react-use';
+import React, { PureComponent } from 'react';
```

Comment:
It was mainly so I could split up rendering in different functions. Granted this could be done by splitting into more components but they're quite small so thought it would just clutter instead.

Reply:
I understand, I'd prefer small functional component over render methods in a larger PureComponent, but that's my preference and we don't really have any clear guidance around this AFAIK, so this is all good. Thank you for answering.